sklearn/tests/test_base.py

```diff
@@ -260,12 +263,54 @@ def test_get_params():
 
 
 def test_is_classifier():
```

Comment:
Since we are quite changing the tests extensively, could we parametrize them. For instance, we would have something like:

```python
@pytest.mark.parametrize(
    "base_estimator, expected_result", [(SVC(), True), (SVR(), False)]
)
def test_is_classifier(estimator, expected_result):
    assert is_classifier(estimator) is expected_result
    assert is_classifier(GridSearchCV(estimator, {"C": [0.1, 1]}))
    assert is_classifier(Pipeline([("estimator", estimator)]))
    assert is_classifier(Pipeline([("estimator_cv", GridSearchCV(estimator, {"C": [0.1, 1]}))])) 
```

Reply:
Good idea. Done.